log.go

```diff
@@ -104,6 +106,24 @@ func (el *eventLogger) EventBegin(ctx context.Context, event string, metadata ..
 	return eip
 }
 
+type contextKey struct{}
+
+var activeEventKey = contextKey{}
```

Comment:
this might not result in a unique value. I would maybe use a string or something?

Reply:
@whyrusleeping it is recommended to use local struct as key for `context.WithValue` to avoid conflicts.

Reply:
https://godoc.org/context#WithValue

Reply:
I agree. However, as a nit, I wouldn't call it `contextKey` as that type shouldn't be reused for other keys. How about `activeEventKeyType` and `activeEventKey`?

Reply:
Good point, I have addressed this in my rebase.